### PR TITLE
ENH: special: `softmax` / `log_softmax` Array API support

### PR DESCRIPTION
**Pull request by crusaderky**
- Add Array API support for `scipy.special.softmax` and `scipy.special.log_softmax`, including no-compute Dask and `jax.jit`
- Refactor unit tests of `log_softmax` to bring them in line with all other tests
- Test `logsumexp` and `_wrap_radians` under no-compute Dask and `jax.jit`

### CODE REVIEW

**Review by j-bowhay — APPROVED** [review]
LGTM, changes are pretty minimal
